### ISSUE TITLE
Chain construction takes a long time

### ISSUE THREAD

**Issue by mohamed82008** [TuringLang/Turing.jl#1066]
Chain construction in Turing currently takes a long time, often way longer than the inference time for large-ish models. Here is an example:
```julia
using Turing
@model high_dim_gauss(D) = begin
    m ~ MvNormal(fill(0.0, D), 1.0)
end
chain = sample(high_dim_gauss(10_000), HMC(0.1, 4), 100)
```
The solution is to probably change the `AxisArray` implementation of MCMCChains to an array of `NamedTuple`s. This will require a change on the Turing side which is only possible after https://github.com/TuringLang/DynamicPPL.jl/issues/4 is fixed. Nothing to be done now but I am filing this issue for reference and discussion.

CC: @cpfiffer, @goedman

**Comment by cpfiffer** [comment]
I've mentioned this on Slack, but I'm very much in support of this. MCMCChains really doesn't lean into the strengths of AxisArrays as much as it should, and in a lot of ways samples from hierarchical models are much better represented in their inherently non-linear form. 

Using just a regular vector of `NamedTuples` is good for people who don't want any of the quick analysis MCMCChains provides, but I'd also like to recommend a [`StructArray`](https://github.com/JuliaArrays/StructArrays.jl) as the actual access and manipulation front end. It'd streamline a lot of stuff. In particular, multivariate draws would become very pleasant to work with:

```julia
@model whatever() = begin
   x ~ MvNormal(30)
end

chain = sample(. . .)
```

If the chain's data storage was a `StructArray`, retrieving the values of `x` is really easy, because you could just call `chain.x` to get a vector of vectors. MCMCChains and Turing has __a ton__ of code to make this work out of the box, and we don't really do it very well.

One solution @mohamed82008 mentioned would be to build a separate package `LightMCMCChains` that inherits from `AbstractMCMC`, and then Turing could kind of slowly phase over to a more streamlined approach to data storage.

**Comment by goedman** [comment]
Interesting idea Cameron. You know my views in AxisArrays within the context of MCMCChains, so I'm certainly in favor of a more accessible approach. StructArrays seems a viable candidate. At one point I used NamedArrays for CmdStan.jl.

On the other hand, building conversions of NamedTuples in for our purposes is not hard.

As I said in an earlier response, for postprocessing (post mcmc diagnostics) I convert to DataFrames (or I've started to look into Particles a bit more).
